lib/json-rpc-server-test/src/main/scala/org/enso/jsonrpc/test/FlakySpec.scala

```
@@ -0,0 +1,22 @@
+package org.enso.jsonrpc.test
+
+import org.scalatest._
+
+/**
+  * Trait is used to mark the tests in the suite as _flaky_ and make them
+  * pass. It changes the behavior for failed tests to return 'pending' instead
+  * of failing the suite.
+  */
+trait FlakySpec extends TestSuite {
+
+  /** Tests that will pass even after a failed run. */
+  def FLAKY_TESTS: Seq[String]
+
+  override def withFixture(test: NoArgTest): Outcome =
```

It would be nicer if you designed a flakyTest tag (like in, ignore). This tag would be responsible for registering flaky tests which failures would be ignored. For example:
```
it should "finish in 3 seconds" flakyTest {
   ...
   testProbe.expectMessagePF {   ...  } 
}
```

---

I do think this approach would be better!

---

Then we need to decide how to implement it. Should the `flackyTest` function just catch all exceptions?

---

Surely you can check the outcome for a given test?

---

It should work the same as `in` tag but additionally, it should register test case as flaky (by adding to a `flakyTests` sequence. The `flakyTests` sequence should be defined in the FlakySpec. The rest is the same as in FlakySpec.scala. Probably you have to define to traits one for FlatSpecs and one for WordSpecs.

---

Nevermind, I found an example in the Scalatest

---

I implemented `flakyTest` similar to [`pendingUntilFixed`](https://github.com/scalatest/scalatest/blob/3.1.x/scalatest/src/main/scala/org/scalatest/Assertions.scala#L1158).